nav-app/src/app/viewmodels.service.ts

```diff
@@ -286,7 +289,16 @@ export class Gradient {
         return tinygradient(colorarray).css('linear', 'to right');
     }
 
-    constructor() { this.setGradientPreset('redgreen'); }
+    constructor() {
+      this.setGradientPreset('redgreen');
+      if (hasCookie("is_user_theme_dark")) this.isDarkTheme = getCookie("is_user_theme_dark") === "true";
+      else this.isDarkTheme = window.matchMedia("(prefers-color-scheme: dark)").matches;
+
+      if (this.isDarkTheme) {
+        this.presets.transparentblue = [new Gcolor("#2e2e3f"), new Gcolor("#66b1ff")];
+        this.presets.transparentred = [new Gcolor("#2e2e3f"), new Gcolor("#ff6666")];
+      }
```

Comment:
- If the user creates a layer tab (and thereby calls the constructor for the gradient) on the dark theme, and then later switches to the light theme, the transparentblue and transparentred presets still use the dark theme settings since they don't re-initialize. This essentially locks the layer into the theme it was created under.
- This hardcodes the dark theme background color, which is a variable in the colors SCSS file. If that file changes due to a future revision it won't be updated here -- a small (but probably unavoidable) maintainability concern.